assets/js/admin/orders/order-overview/_refund.js

```diff
@@ -59,74 +59,104 @@ $( document.body ).on( 'click', '.ui-widget-overlay', function ( e ) {
+	if ( $( this ).hasClass( 'edd-order-item-refund-quantity' ) ) {
+		let subtotalField = parent.find( '.edd-order-item-refund-subtotal' ),
+			taxField = parent.find( '.edd-order-item-refund-tax' ),
+			originalSubtotal = parseFloat( subtotalField.data( 'original' ) ),
+			originalTax = parseFloat( taxField.data( 'original' ) ),
+			originalQuantity = parseInt( quantityField.attr( 'max' ) ),
+			calculatedSubtotal = ( originalSubtotal / originalQuantity ) * quantity,
+			calculatedTax = ( originalTax / originalQuantity ) * quantity;
+
+		// Make sure totals don't go over maximums.
+		if ( calculatedSubtotal > parseFloat( subtotalField.data( 'max' ) ) ) {
+			calculatedSubtotal = subtotalField.data( 'max' );
+		}
+		if ( calculatedTax > parseFloat( taxField.data( 'max' ) ) ) {
+			calculatedTax = taxField.data( 'max' );
+		}
+
+		// Guess the subtotal and tax for the selected quantity.
+		subtotalField.val( parseFloat( calculatedSubtotal ).toFixed( edd_vars.currency_decimals ) );
+		taxField.val( parseFloat( calculatedTax ).toFixed( edd_vars.currency_decimals ) );
+	}
+
+	recalculateRefundTotal();
+} );
+
+/**
+ * Calculates all the final refund values.
+ */
+function recalculateRefundTotal() {
+	let newSubtotal   = 0,
+		newTax        = 0,
+		newTotal      = 0,
+		canRefund     = false,
+		allInputBoxes = $( '#edd-refund-order-dialog .edd-order-item-refund-input' );
 
 	// Set a readonly while we recalculate, to avoid race conditions in the browser.
-	all_checkboxes.prop('readonly', true);
-	$('#edd-refund-submit-button-wrapper .spinner').css('visibility', 'visible');
+	allInputBoxes.prop( 'readonly', true );
+	$( '#edd-refund-submit-button-wrapper .spinner' ).css( 'visibility', 'visible' );
 
-	$('#edd-refund-order-dialog tbody .check-column input[type="checkbox"]:checked').each( function() {
-		let item_parent = $(this).parent().parent();
+	// Loop over all order items.
+	$( '#edd-refund-order-dialog .edd-order-item-refund-quantity' ).each( function() {
+		let thisItemQuantity = parseInt( $( this ).val() );
+
+		if ( ! thisItemQuantity ) {
+			return;
+		}
+
+		let thisItemParent = $( this ).parent().parent();
 
 		// Values for this item.
-		let item_amount   = parseFloat( item_parent.find('span[data-amount]').data('amount') ),
-			item_tax      = parseFloat( item_parent.find('span[data-tax]').data('tax') ),
-			item_total    = parseFloat( item_parent.find('span[data-total]').data('total') ),
-			item_quantity = parseInt( item_parent.find('.column-quantity').text() );
-
-		new_subtotal += item_amount;
-		new_tax      += item_tax;
-		new_total    += item_total;
-		can_refund    = true;
-	});
+		let thisItemSubtotal = 0,
+			thisItemTax      = 0,
+			thisItemTotal    = 0;
+
+		if ( thisItemQuantity ) {
+			thisItemSubtotal = parseFloat( thisItemParent.find( '.edd-order-item-refund-subtotal' ).val() ),
+			thisItemTax      = parseFloat( thisItemParent.find( '.edd-order-item-refund-tax' ).val() ),
```

Comment:
If taxes are not enabled, this line results in `NaN` as the input is not present; it borks the subtotal and total calculations.